invenio_records_resources/references/entity_resolvers/results.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 Graz University of Technology.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Resolver for result items."""
+
+from invenio_access.permissions import system_identity
+
+from .base import EntityProxy, EntityResolver
+
+
+class ServiceResultProxy(EntityProxy):
+    """Resolver proxy for a result item entity."""
+
+    def __init__(self, resolver, ref_dict, service):
+        """Constructor."""
+        super().__init__(resolver, ref_dict)
+        self.service = service
+
+    def _resolve(self):
+        """Resolve the result item from the proxy's reference dict."""
+        id_ = self._parse_ref_dict_id()
+        # TODO: Make identity customizable
+        return self.service.read(system_identity, id_).to_dict()
+
+    def get_needs(self, ctx=None):
+        """Return needs."""
+        return []
+
+    def pick_resolved_fields(self, identity, resolved_dict):
+        """Select which fields to return when resolving the reference."""
+        return {"id": resolved_dict["id"]}
+
+
+class ServiceResultResolver(EntityResolver):
+    """Resolver for result items."""
+
+    def __init__(
+        self,
+        service_id,
+        type_key,
+        proxy_cls=ServiceResultProxy,
+        item_cls=None,
+        record_cls=None,
+    ):
+        """Constructor.
+
+        :param item_cls: The result item class to use.
+        :param service_id: The record service id.
+        :param type_key: The value to use for the TYPE part of the ref_dicts.
+        """
+        super().__init__(service_id)
+        self.type_key = type_key
+        self.proxy_cls = proxy_cls
+        self.item_cls = item_cls or self.get_service().config.result_item_cls
+        self.record_cls = record_cls or self.get_service().record_cls
```

Comment:
```suggestion
        self.item_cls = item_cls or self.get_service().config.result_item_cls
        self.record_cls = record_cls or self.get_service().record_cls
```

minor: we should compute these "lazily" either in the functions they're used in or via properties, since depending on when we initialize the `ServiceResultResolver` not all services/registries will be initialized.